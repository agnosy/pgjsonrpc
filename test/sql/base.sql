\set ECHO 0
BEGIN;
\i sql/pgjsonrpc.sql
\set ECHO all

-- You should write your tests

SELECT pgjsonrpc('foo', 'bar');

SELECT 'foo' #? 'bar' AS arrowop;

CREATE TABLE ab (
    a_field pgjsonrpc
);

INSERT INTO ab VALUES('foo' #? 'bar');
SELECT (a_field).a, (a_field).b FROM ab;

SELECT (pgjsonrpc('foo', 'bar')).a;
SELECT (pgjsonrpc('foo', 'bar')).b;

SELECT ('foo' #? 'bar').a;
SELECT ('foo' #? 'bar').b;

ROLLBACK;
