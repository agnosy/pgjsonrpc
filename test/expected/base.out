\set ECHO none
-- You should write your tests
SELECT jsonrpc.execute('{"id":1, "method":"test"}');
                                           execute                                            
----------------------------------------------------------------------------------------------
 {"id" : "1", "jsonrpc" : "2.0", "error" : {"code" : -32601, "message" : "Method not found"}}
(1 row)

SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echo", "params": {"message": "Hello jsonrpc!"}}');
                           execute                            
--------------------------------------------------------------
 {"id" : "1", "jsonrpc" : "2.0", "result" : "Hello jsonrpc!"}
(1 row)

INSERT INTO jsonrpc.methods(name, description, function_name)
VALUES('echotest', 'Echoes the parameter passed to echo.', 'jsonrpc.echo');
SELECT jsonrpc.execute('{"jsonrpc": "2.0", "id":1, "method":"echotest", "params": {"message": "Hello jsonrpc!"}}');
                           execute                            
--------------------------------------------------------------
 {"id" : "1", "jsonrpc" : "2.0", "result" : "Hello jsonrpc!"}
(1 row)

ROLLBACK;
